pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -345,6 +348,68 @@ public SuccessResponse updateTableConfig(
 
     return new SuccessResponse("Table config updated for " + tableName);
   }
+  
+  /**
+   * Truncate table, delete all contents of table without removing table configuration and schema. 
```

Comment:
Let's add here that for Truncate for REALTIME includes creating new CONSUMING segments in segment metadata and ideal state